docs/styleguide/generic.rst

```diff
@@ -162,3 +162,52 @@ Comments
       do_something()
 
 * Avoid end of line comments.
+
+Git
+===
+
+Feature branch development
+--------------------------
+
+The code of our project is shared by all members of the team.
+
+We often have a feature branch that is driven by a single developer, but during
+the branch lifetime or the review process that branch may get contributions from
+other developers. For example typo-fixes or small GUI fixes.
+
+To reduce conflicts and confusion between colleagues, avoid using git commands
+that rewrite the history of commits you have already pushed (such as rebase or
+commit with amend).
+
+See  :ref:`here on how to keep a feature branch up to date
+<keep-feature-branch-up-to-date>`.
+
+Don't revert, just push the fix
+-------------------------------
+
+If you find an issue after merging into the main branch, avoid the trouble of
+reverting the commit before pushing a fix.
+
+If it is clear that the commit is broken (like showing a failing test), and the
+team will not release it, then feel free to just push the fix on top.
```

Comment:
This should be removed. For me is confustion.

The idea is that you don't just push the fix on top. You will create a branch, fix it in the branch, as a review and then merge in the main branch.